CRM/Admin/Form/Setting.php

```diff
@@ -96,9 +97,13 @@ public function buildQuickForm() {
     );
 
     $descriptions = array();
-    foreach ($this->_settings as $setting => $group) {
-      $settingMetaData = civicrm_api('setting', 'getfields', array('version' => 3, 'name' => $setting));
-      $props = $settingMetaData['values'][$setting];
+    // Getting all once seems better than multiple calls.
+    $allSettingMetaData = civicrm_api3('setting', 'getfields', array());
```

Comment:
probably better for this to move to seperate method : 

```php

```php
$settingMetaData = $this->getSettingsMetaData();

```php
/**
* Docblock goes here
*/
private function getSettingsMetaData() {
    $allSettingsMetaData = civicrm_api3('setting', 'getfields', array());

    $settingMetaData = array_intersect_key($allSettingMetaData['values'], $this->_settings);
    // re-ordering the settings to follow $this->_settings order.
    $settingMetaData = array_merge($this->_settings, $settingMetaData);
   
    return $settingMetaData ;
}



```